src/test/java/com/training/web/tests/flipkart/PaymentPageTests.java

```diff
@@ -0,0 +1,44 @@
+package com.training.web.tests.flipkart;
+
+import com.training.basetest.WebBaseTest;
+import com.training.web.pages.flipkart.FlipkartHomePage;
+import com.training.web.pages.flipkart.PaymentPage;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+
+import static com.codeborne.selenide.Selenide.open;
+
+public class PaymentPageTests extends WebBaseTest {
+    PaymentPage paymentPage;
+
+    @BeforeEach
+    public void startup() {
+        open("https://www.flipkart.com/");
+        paymentPage = new FlipkartHomePage().closePopup().clickPayments();
+    }
+
+    @Test
+    public void testPaymentPageNavigation() {
+
+        /**
+         * Assertiong the test has navigated to payments page
+         */
+        softAssertions.assertThat(paymentPage.getUrl()).as("Test didn't navigate to payment page")
+                .isEqualTo("https://www.flipkart.com/pages/payments");
```

Comment:
You should use hard assert here,  bcoz if you not able to retrive the url or url doesnt match then the program should fail here its self